src/custom-select/index.ts

```diff
@@ -29,30 +29,33 @@ class CustomSelect {
     }
   }
 
-  onClickDeactivate(event: MouseEvent & Tag.Event) {
+  onHoverDeactivate(event: MouseEvent & Tag.Event) {
     event.preventUpdate = true;
     if (!this.props.hover && !this.isActive) {
       this.update();
     }
   }
 
-  onLostFocus = (event: FocusEvent & Tag.Event) => {
+  onClickDeactivate = (event: MouseEvent & Tag.Event) => {
     event.preventUpdate = true;
-    if (!this.props.hover) {
-      this.isActive = false;
+    if (!this.props.hover && event.target !== this.refs.toggle.refs.button) {
+      utils.WINDOW.document().removeEventListener('click', this.onClickDeactivate);
```

Comment:
The one change I'd make is that rather than explicitly calling remove or add in the actual event handlers, add an `onUpdate()` in which you can check the updated state (`state.isActive`) to see if you should be adding or removing the handler.